src/ProtocolV3TestBase.sol

```diff
@@ -596,22 +608,125 @@ contract ProtocolV3TestBase is CommonTestBase {
     address user,
-    uint256 amount,
-    bool receiveAToken
+    uint256 amount
   ) internal {
-    deal2(debt.underlying, liquidator, amount);
+    deal2(borrow.underlying, liquidator, amount);
+
+    address debtToken = borrow.variableDebtToken;
 
     vm.startPrank(liquidator);
-    IERC20(debt.underlying).approve(address(pool), amount);
-    console.log('LIQUIDATE: Collateral: %s, Debt: %s, Debt Amount: %s', collateral.symbol, debt.symbol, _formattedAmount(amount, debt.decimals));
-    pool.liquidationCall(collateral.underlying, debt.underlying, user, amount, receiveAToken);
+    IERC20(borrow.underlying).approve(address(pool), amount);
+
+    LiquidationBalanceAssertions memory balances;
+
+    balances.aTokenBorrowerBefore = IERC20(collateral.aToken).balanceOf(user);
+    balances.aTokenTreasuryBefore = IERC20(collateral.aToken).balanceOf(IAToken(collateral.aToken).RESERVE_TREASURY_ADDRESS());
+
+    balances.collateralATokenBefore     = IERC20(collateral.underlying).balanceOf(collateral.aToken);
+    balances.collateralLiquidatorBefore = IERC20(collateral.underlying).balanceOf(liquidator);
+
+    balances.debtBefore = IERC20(debtToken).balanceOf(user);
+
+    balances.borrowATokenBefore     = IERC20(borrow.underlying).balanceOf(borrow.aToken);
+    balances.borrowLiquidatorBefore = IERC20(borrow.underlying).balanceOf(liquidator);
+
+    // TODO: Add totalSupply assertions
+
+    console.log('LIQUIDATE: Collateral: %s, Debt: %s, Debt Amount: %s', collateral.symbol, borrow.symbol, _formattedAmount(amount, borrow.decimals));
+    pool.liquidationCall(collateral.underlying, borrow.underlying, user, amount, false);
     vm.stopPrank();
+
+    ( uint256 totalCollateralToLiquidate, uint256 amountToProtocol )
+      = _getLiquidationAmounts(collateral, borrow, pool, balances.debtBefore);
+
+    balances.aTokenBorrowerAfter = IERC20(collateral.aToken).balanceOf(user);
+    balances.aTokenTreasuryAfter = IERC20(collateral.aToken).balanceOf(IAToken(collateral.aToken).RESERVE_TREASURY_ADDRESS());
+
+    balances.collateralATokenAfter     = IERC20(collateral.underlying).balanceOf(collateral.aToken);
+    balances.collateralLiquidatorAfter = IERC20(collateral.underlying).balanceOf(liquidator);
+
+    balances.debtAfter = IERC20(debtToken).balanceOf(user);
+
+    balances.borrowATokenAfter     = IERC20(borrow.underlying).balanceOf(borrow.aToken);
+    balances.borrowLiquidatorAfter = IERC20(borrow.underlying).balanceOf(liquidator);
+
+    assertEq(balances.debtAfter, 0);  // All debt removed, full liquidation
+
+    // Checks to ensure diffs aren't zero
+    assertLt(balances.collateralATokenAfter,     balances.collateralATokenBefore);      // Collateral balance of aToken decreases
+    assertGt(balances.collateralLiquidatorAfter, balances.collateralLiquidatorBefore);  // Liquidator receives collateral
+    assertLt(balances.aTokenBorrowerAfter,       balances.aTokenBorrowerBefore);        // Collateral removed from aToken liquidity
```

Comment:
These are here to protect against a situation where both values are actually zero but aren't expected to be, since these values aren't being explicitly asserted its good to put in these guard rails against unexpected behaviours.